### PR TITLE
amfora 1.8.0 (new formula)

### PR DESCRIPTION
**Pull request by vladimyr**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

### CODE REVIEW

File: Formula/amfora.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  url "https://github.com/makeworld-the-better-one/amfora.git",
+      tag:      "v1.8.0",
+      revision: "71385e9f4e91ab982076856cac95320173ccea73"
+  license "GPL-3.0-or-later"
+  head "https://github.com/makeworld-the-better-one/amfora.git"
+
+  depends_on "go" => :build
+
+  def install
+    ENV["GO111MODULE"] = "on"
```
Comment (chenrui333) [inline Formula/amfora.rb:13]:
```suggestion
```

With go1.16, go modules is turned on by default.

File: Formula/amfora.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    ldflags = %W[
+      -s -w
+      -X main.version=#{version}
+      -X main.commit=#{Utils.git_head}
+      -X main.builtBy=homebrew
+    ]
+    system "go", "build", "-ldflags", ldflags.join(" "), *std_go_args
```
Comment (chenrui333) [inline Formula/amfora.rb:20]:
```suggestion
    system "go", "build", *std_go_args(ldflags: ldflags.join(" "))
```

File: Formula/amfora.rb (hunk 1 above)
Comment (vladimyr) [reply Formula/amfora.rb]: Golang is obviously not my strongest skill :D

File: Formula/amfora.rb [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  depends_on "go" => :build
+
+  def install
+    ldflags = %W[
+      -s -w
+      -X main.version=#{version}
+      -X main.commit=#{Utils.git_head}
+      -X main.builtBy=homebrew
+    ]
+    system "go", "build", "-ldflags", *std_go_args(ldflags: ldflags.join(" "))
```
Comment (carlocab) [inline Formula/amfora.rb:19]:
```suggestion
    system "go", "build", *std_go_args(ldflags: ldflags.join(" "))
```

Comment (vladimyr) [reply Formula/amfora.rb:19]: Sorry 🤦

File: Formula/amfora.rb [hunk 4]
```diff
@@ -0,0 +1,37 @@
+class Amfora < Formula
+  desc "Fancy terminal browser for the Gemini protocol"
+  homepage "https://github.com/makeworld-the-better-one/amfora"
+  url "https://github.com/makeworld-the-better-one/amfora.git",
+      tag:      "v1.8.0",
+      revision: "71385e9f4e91ab982076856cac95320173ccea73"
+  license "GPL-3.0-or-later"
```
Comment (nandahkrishna) [inline Formula/amfora.rb:7]: Should this be `GPL-3.0-only`? I find no mentions of "or later" in non-template text (see [search query](https://github.com/makeworld-the-better-one/amfora/search?q=or+later)).

Comment (chenrui333) [reply Formula/amfora.rb:7]: here is the debian copyright file reference, https://metadata.ftp-master.debian.org/changelogs//main/a/amfora/amfora_1.8.0-1_copyright

Comment (chenrui333) [reply Formula/amfora.rb:7]: I think we can go with GPL-3.0-or-later, here is the arch side ref:

Comment (carlocab) [reply Formula/amfora.rb:7]: Debian doesn't indicate `-or-later`, though, and I'm inclined to trust them here. It seems to actually be
```ruby
license all_of: [
  "GPL-3.0-only"
  any_of: ["GPL-3.0-only", "MIT"]
]
```

Comment (vladimyr) [reply Formula/amfora.rb:7]: Only [`rr` code](https://github.com/makeworld-the-better-one/amfora/tree/master/rr#license) (amfora's subpackage) is dual-licensed under GPL3 (only?, inherits amfora's license) or MIT. According to this, I think we should stick with `GPL-3.0-only`? 🤔

Comment (nandahkrishna) [reply Formula/amfora.rb:7]: We can specify with comments which licenses apply to which directories/files, like in `e2fsprogs`.

Comment (vladimyr) [reply Formula/amfora.rb:7]: > We can specify with comments which licenses apply to which directories/files, like in `e2fsprogs`.

Interesting, thanks for the pointer! 👍

Comment (carlocab) [reply Formula/amfora.rb:7]: > Only [`rr` code](https://github.com/makeworld-the-better-one/amfora/tree/master/rr#license) (amfora's subpackage) is dual-licensed under GPL3 (only?, inherits amfora's license) or MIT. According to this, I think we should stick with `GPL-3.0-only`? 🤔

This is exactly why you can either license it under GPLv3 entirely or both MIT (rr subpackage) and GPLv3 (everything else).

Comment (chenrui333) [reply Formula/amfora.rb]: we should definitely have some license hell emoji :D

**Review by nandahkrishna — APPROVED** [review]
LGTM, thanks @vladimyr!

**Pull request closed without merging**